src/legacy/ui/ui_render/bootstrap/template.js.hbs

```diff
@@ -68,6 +68,24 @@ if (window.__kbnStrictCsp__ && window.__kbnCspNotEnforced__) {
       });
     }
 
+    // When rendering in Kibana Platform mode, only serve the assets needed by Core.
+    if ('{{appId}}' === 'core') {
```

Comment:
I think we should do the condition in the template and not in the JS. If we define `isNewPlatformPlugin` in the template variables we can just do:

```hbs
{{!-- When rendering in Kibana Platform mode, only serve the assets needed by Core. --}}
{{#if isNewPlatformPlugin}}
load([
  {{#each sharedJsDepFilenames}}
    '{{../regularBundlePath}}/kbn-ui-shared-deps/{{this}}',
  {{/each}}
  '{{regularBundlePath}}/kbn-ui-shared-deps/{{sharedJsFilename}}',
  '{{regularBundlePath}}/plugin/kibanaUtils/kibanaUtils.plugin.js',
  '{{regularBundlePath}}/plugin/esUiShared/esUiShared.plugin.js',
  '{{regularBundlePath}}/plugin/kibanaReact/kibanaReact.plugin.js',
  '{{regularBundlePath}}/core/core.entry.js',
  {{#each styleSheetPaths}}
    '{{this}}',
  {{/each}}
]);
{{else}}
load([
  {{#each sharedJsDepFilenames}}
    '{{../regularBundlePath}}/kbn-ui-shared-deps/{{this}}',
  {{/each}}
  '{{regularBundlePath}}/kbn-ui-shared-deps/{{sharedJsFilename}}',
  '{{dllBundlePath}}/vendors_runtime.bundle.dll.js',
  {{#each dllJsChunks}}
    '{{this}}',
  {{/each}}
  '{{regularBundlePath}}/commons.bundle.js',
  '{{regularBundlePath}}/plugin/kibanaUtils/kibanaUtils.plugin.js',
  '{{regularBundlePath}}/plugin/esUiShared/esUiShared.plugin.js',
  '{{regularBundlePath}}/plugin/kibanaReact/kibanaReact.plugin.js',
  '{{regularBundlePath}}/{{appId}}.bundle.js',
  {{#each styleSheetPaths}}
    '{{this}}',
  {{/each}}
]);
{{/if}}
```

Reply:
I actually decided to move all of the string building logic into the response handler code. It makes the template much simpler to understand.